code/modules/mob/living/carbon/carbon.dm

```diff
@@ -287,17 +289,42 @@
 				if(H.bg)
 					to_chat(H, span("warning", "You sense some disturbance to your physical body, like someone is trying to wake you up."))
 				else
-					M.visible_message("<span class='notice'>[M] shakes [src] trying to wake [t_him] up!</span>", \
-										"<span class='notice'>You shake [src], but they do not respond... Maybe they have S.S.D?</span>")
-			else if(lying || src.sleeping)
-				src.sleeping = max(0,src.sleeping-5)
-				if(src.sleeping == 0)
-					src.resting = 0
-				M.visible_message("<span class='notice'>[M] shakes [src] trying to wake [t_him] up!</span>", \
-									"<span class='notice'>You shake [src] trying to wake [t_him] up!</span>")
+					M.visible_message(span("notice", "[M] shakes [src] trying to wake [t_him] up!"), \
+										span("notice", "You shake [src], but they do not respond... Maybe they have S.S.D?"))
+			else if(lying)
+				if(src.sleeping)
+					src.sleeping = max(0,src.sleeping-5)
+					M.visible_message(span("notice", "[M] shakes [src] trying to wake [t_him] up!"), \
+										span("notice", "You shake [src] trying to wake [t_him] up!"))
+				else
+					M.help_up_offer = !M.help_up_offer
+					if(M.help_up_offer)
+						M.visible_message(span("notice", "[M] holds a hand out to [src]."), \
+											span("notice", "You hold a hand out to [src]."))
+					else
+						M.visible_message(span("warning", "[M] retracts their hand from [src]'s direction."), \
+											span("warning", "You retract your hand from [src]'s direction."))
```

Comment:
The fact that this uses the M mob's variable is odd. What happens when 2 people offer the same person help? Basically, first one will get `You hold a hand out to [src].` and the other one will get `You retract your hand from [src]'s direction.`, despite the other one not having offered help in the first place.

It should either use the `src` variable of it, or store a reference to the mob offering help.

Reply:
![image](https://user-images.githubusercontent.com/3950115/70843508-70c81380-1de8-11ea-8925-c6b6830e64af.png)

I'm not really sure this seems to be case, though I apologise if I may have misunderstood what you're pointing out.

What DOES happen, though, is that if the person on the ground takes one of the people's hands, the other person remains holding their hand out. This is reset to 0 though as soon as that other person does any action to automatically reset it to 0, such as by moving.